app/src/main/java/ch/epfl/sdp/game/Game.java

```diff
@@ -108,11 +114,42 @@ public void destroyGame() {
     public void update() {
         itUpdatable = updatables.iterator();
 
+        updateLocalScoreOfPlayers();
+
         while (itUpdatable.hasNext()) {
             itUpdatable.next().update();
         }
     }
 
+    /**
+     * This method update the local score of all the player in the game (the rule is that every 10 seconds, a player gets 10 points
+     * and if he walked more than 10 meters, he also gets 10 points
+     */
+    private void updateLocalScoreOfPlayers() {
+        if (numberOfUpdates > 9 * gameThread.getFPS()) {
+            numberOfUpdates = 0;
+            for (Player player : PlayerManager.getPlayers()) {
+                player.updateLocalScore();
+            }
+        } else {
+            numberOfUpdates++;
```

Comment:
If I'm not mistaken FPS is frames per second and you are doing (9 seconds)*(frames per seconds) = number of frames in 9 seconds. If the unit of measurement of numberOfUpdates is in frames then you should here numberOfUpdates += (frames per second)

Reply:
I know it's weird, but it's the only way I got timing to work correctly. That's why I thought something was wrong somewhere